### PR TITLE
Fix: propagate --cache-from to buildx build

### PR DESCRIPTION
**Pull request by dhhyi**
I am trying to use the new `--cache-to` feature from #570 with a [registry cache](https://docs.docker.com/build/cache/backends/registry/).

If I understand it correctly this only works when using both `--cache-to` and `--cache-from`, but the devcontainers-cli does not send `--cache-from` to `docker buildx build`.

### CODE REVIEW

File: src/spec-node/singleContainer.ts [hunk 1]
```diff
@@ -202,6 +202,9 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 		if (buildParams.buildxCacheTo) {
 			args.push('--cache-to', buildParams.buildxCacheTo);
 		}
+		if (buildParams.additionalCacheFroms) {
+			buildParams.additionalCacheFroms.forEach(cacheFrom => args.push('--cache-from', cacheFrom));
```
Comment (chrmarti) [inline src/spec-node/singleContainer.ts:206]: Line 231 below already does this. 🤔

File: src/spec-node/containerFeatures.ts [hunk 2]
```diff
@@ -80,6 +80,9 @@ export async function extendImage(params: DockerResolverParameters, config: Subs
 		if (params.buildxCacheTo) {
 			args.push('--cache-to', params.buildxCacheTo);
 		}
+		if (params.additionalCacheFroms) {
```
Comment (chrmarti) [inline src/spec-node/containerFeatures.ts:83]: Skip if `params.buildNoCache` is set.

**Review by chrmarti — CHANGES_REQUESTED** [review]
Thanks for the PR! Left a few comments.

File: src/spec-node/containerFeatures.ts (hunk 2 above)
Comment (chrmarti) [reply src/spec-node/containerFeatures.ts:83]: Just noticed `params.additionalCacheFroms` is always set, no need to check for that.

File: src/spec-node/singleContainer.ts (hunk 1 above)
Comment (dhhyi) [reply src/spec-node/singleContainer.ts]: hmm, I don't remember why I added it, sorry. It also seems to be working without. 🤷

File: src/spec-node/containerFeatures.ts (hunk 2 above)
Comment (dhhyi) [reply src/spec-node/containerFeatures.ts]: ✅

**Review by chrmarti — APPROVED** [review]
Thanks!

**Review by samruddhikhandale — APPROVED** [review]
Looks good, thanks!

**Pull request merged**